src/main/java/net/rubygrapefruit/platform/file/FileWatchEvent.java

```diff
@@ -4,35 +4,21 @@
 
 public interface FileWatchEvent {
 
-    /**
-     * The type of the change. When {@link Type#FAILURE}, {@link #getFailure()} returns
-     * the failure, and {@link #getPath()} returns {@code null}.
-     * Otherwise {@link #getFailure()} returns {@code null} and {@link #getPath()}
-     * returns either the path of the change, or {@code null} if a path is not known.
-     */
-    Type getType();
+    void handleEvent(Handler handler);
 
-    /**
-     * The path that has been changed. Can be {@code null} when {@link #getType()} is
-     * {@link Type#FAILURE} or {@link Type#UNKNOWN}.
-     *
-     * See notes on the {@link FileWatcher} implementation for:
-     *
-     * <ul>
-     *     <li>whether calls to this method can be expected from a single thread or multiple different ones,</li>
-     *     <li>how actual events are reported.</li>
-     * </ul>
-     */
-    @Nullable
-    String getPath();
+    interface Handler {
+        void handleChangeEvent(ChangeType type, String absolutePath);
```

Comment:
```suggestion
        void handleChange(ChangeType type, String absolutePath);
```

I would remove the `Event` from the `handle*` methods.